tests/test_raster.py

```diff
@@ -3204,7 +3232,7 @@ def test_reproject(self, mask: gu.Mask) -> None:
             match="Reprojecting a mask with a resampling method other than 'nearest', "
             "the boolean array will be converted to float during interpolation.",
         ):
-            mask.reproject(resampling="bilinear")
+            mask.reproject(res=50, resampling="bilinear", force_source_nodata=2)
```

Comment:
why was a change needed here?

Reply:
I am not sure, but the wrong `UserWarning` was raised (the "returning self" one because nothing changed for this reprojection). And looking at the test, this makes sense. So the question is why did it pass in the first place...

Reply:
The test should be more robust now, I didn't get the courage to investigate more